### DISCUSSION TITLE
Roadmap discussion for v9

### DISCUSSION THREAD

**Discussion by BobbieGoede** [nuxt-modules/i18n#2814] [General]
> [!NOTE]  
> This list is incomplete and a work in progress, feel free to suggest features, topics or issues that you think should be included in v9.

There's still plenty of issues open for v8 and we'll continue to work on fixing those, this roadmap is simply to plan and discuss changes that would be breaking. There is currently no estimate or deadline for the next major version, you don't have to worry about migrating (again) anytime soon.

### Features
- [ ] support vue-i18n v10
- [ ] support `v-t` fully SSR 

### Breaking changes
- [ ] https://github.com/nuxt-modules/i18n/pull/2801
- [ ] Rename locales property `iso` to `language`
  - [ ] Add alias as described in https://github.com/nuxt-modules/i18n/issues/2449#issuecomment-1740517847 to allow for easier transition
- [ ] https://github.com/nuxt-modules/i18n/issues/2712
- [ ] Change route localization to omit parent from path in `defineI18nRoute` as described in https://github.com/nuxt-modules/i18n/pull/2743#issue-2108024664
- [ ] https://github.com/nuxt-modules/i18n/issues/1985
- [ ] Restructure `runtimeConfig` keys
  - [ ] Rename `locales` to `domainLocales` and `configLocales` to `locales` (https://github.com/nuxt-modules/i18n/pull/2828#discussion_r1513740467)
- [ ] enable jit compilation as default (remove `jit` option) https://i18n.nuxtjs.org/docs/options/compilation#jit

### Considering
* Changing/aligning `useLocaleHead` and `useSetI18nParams` argument defaults
* Change `locales` configuration
  * Remove `file`, only accepting `files` array instead, simplifies options merging.
  * Configure locale file paths using absolute paths (using `resolve`), removing `langDir` options. This would simplify layer option merging and be consistent with module hooks.

**Comment by kazupon** [comment]
We should work to flat nuxt i18n and vue-i18n config 
https://github.com/nuxt-modules/i18n/issues/1985

**Reply by luca-smartpricing** [reply]
Maybe you could consider removing the auto-complete alternate langs as discussed here: #2784

**Comment by s00d** [comment]
Hello, let's add these changes to the new version https://github.com/nuxt-modules/i18n/pull/2937 or at least discuss them.

**Comment by BobbieGoede** [comment]
Closing this discussion as we're approaching v9's release, non-breaking changes are of course still on the table and tracked in issues.
